gulpfile.js

```diff
@@ -545,20 +545,36 @@ async function pruneScriptsForZip(packageJsonPath) {
   return gulp.src(noPreparePackageJson).pipe(gulpRename(packageJsonPath));
 }
 
-export const postversion = function () {
+export const postversion = async function () {
   const workspace = argv.workspace;
   if (!workspace) {
     return;
   }
-
-  // After the workspace version is bumped by `npm version`,
-  // update the root `package.json` file to depend on the new version
   const directory = workspace.replaceAll(`@${scope}/`, ``);
   const workspacePackageJson = require(`./packages/${directory}/package.json`);
   const version = workspacePackageJson.version;
 
-  packageJson.dependencies[workspace] = version;
-  return writeFile("package.json", JSON.stringify(packageJson, undefined, 2));
+  // Iterate through all package JSONs that may depend on the updated package and
+  // update the version of the updated workspace.
+  const packageJsons = await globby([
+    "./package.json",
+    "./packages/*/package.json",
+  ]);
+  const promises = packageJsons.map(async (packageJsonPath) => {
+    // Ensure that we don't check the updated workspace itself.
+    if (basename(dirname(packageJsonPath)) === directory) {
+      return Promise.resolve();
+    }
+    // Ensure that we only update workspaces where the dependency to the updated workspace already exists.
+    const packageJson = require(packageJsonPath);
+    if (!(workspace in packageJson.dependencies)) {
```

Comment:
As dependencies are ultimately pretty arbitrary key/value pairs, I think this may be an [undesirable use of `in`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in#inherited_properties) and we should opt for `Object.hasOwn` instead.